renderer_test.go

```diff
@@ -0,0 +1,190 @@
+package atlassianwikirenderer
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"github.com/yuin/goldmark"
+)
+
+type testCase struct {
+	input    string
+	expected string
+}
+
+func TestTaskCheckbox(t *testing.T) {
+	runTest(t, testCase{
+		input:    "- [ ] Task 1\n- [x] Task 2 (completed)\n- [ ] Task 3",
+		expected: "*  [ ] Task 1\n*  [x] Task 2 (completed)\n*  [ ] Task 3",
+	})
+}
+
+func TestHeadingRendering(t *testing.T) {
+	runTest(t, testCase{
+		input:    "# Heading 1",
+		expected: "h1.Heading 1",
+	})
+}
+
+func TestBlockquoteRendering(t *testing.T) {
+	runTest(t, testCase{
+		input:    "> This is a blockquote",
+		expected: "{quote}\nThis is a blockquote\n{quote}",
+	})
+}
+
+func TestFencedCodeBlockRendering(t *testing.T) {
+	runTest(t, testCase{
+		input:    "```This is a fenced code block```",
+		expected: "{{This is a fenced code block}}",
+	})
+}
+
+func TestParagraphRendering(t *testing.T) {
+	runTest(t, testCase{
+		input:    "This is a paragraph. It can contain text,\nline breaks, and various formatting options.\n\nYou can separate paragraphs by leaving a blank line between them.",
```

Comment:
`various formatting options` -- Add some bold, underline and italics to this paragraph to illustrate this.